trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1272,6 +1361,33 @@ where
         }
     }
 
+    /// Provide the requested content key and content value for the acceptor
+    fn provide_requested_content(
+        &self,
+        response: &Accept,
+        content_keys_offered: Vec<TContentKey>,
+    ) -> Vec<ByteList> {
+        let mut content_items: Vec<ByteList> = Vec::new();
+
+        for (i, key) in response
+            .content_keys
+            .clone()
+            .iter()
+            .zip(content_keys_offered.iter())
+        {
+            if i == true {
+                match self.storage.read().get(&key.clone()) {
+                    Ok(content) => match content {
+                        Some(content) => content_items.push(content.into()),
+                        None => {}
```

Comment:
Since we offered this content, a `None` is a bit of a surprise, right? I'm thinking it's worth a log here. Maybe even a `warn!`.

Also, on the receiving side, if a peer skips over some content that we accepted, I think we (and others) might cause a validation failure. So at that point, it might even just be fine to give up altogether and stop collecting items. But I don't have a strong intuition about that yet.

I guess that would change the returned type of this method into a `Result`, and return an `Err` if there was a failure to collect any of the content.

I guess all of this applies to getting an `Err` on the `get()`, also. (except that we already have a `warn!`)